records_mover/records/delimited/utils.py

```diff
@@ -1,15 +1,19 @@
-from .types import RecordsHints
+from .types import RecordsHints, UntypedRecordsHints
+from .validated_records_hints import ValidatedRecordsHints
 import logging
-from typing import Iterable
+from typing import Iterable, Union
 
 
 logger = logging.getLogger(__name__)
 
 
 def complain_on_unhandled_hints(fail_if_dont_understand: bool,
                                 unhandled_hints: Iterable[str],
-                                hints: RecordsHints) -> None:
-    unhandled_bindings = [f"{k}={hints[k]}" for k in unhandled_hints]
+                                hints: Union[RecordsHints,
+                                             UntypedRecordsHints,
+                                             ValidatedRecordsHints]) -> None:
+    # TODO: this probably doesn't handle ValidatedRecordsHints yet
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker